### DISCUSSION TITLE
How would i make this lib portable? (single/multi core boards)

### DISCUSSION THREAD

**Discussion by GM-Script-Writer-62850** [micropython/micropython#15974] [General]
I made a lib multicore safe, but I assume `_thread` is not present on single core boards so importing it would cause a error

from `_thread` i am using `allocate_lock` and using `acquire`/`release` on the lock

* I am using a lock to make sure 2 or more cores do not try to write over SPI at the same time as doing that would cause unexpected results

is this what i should do? so it will also run a single core board?
```
try:
    from _thread import allocate_lock
except:
    class allocate_lock:
        def acquire():
            return
        def release():
            return
```

**Comment by peterhinch** [comment]
That looks OK. Note that `_thread` may be present on single core boards. For example there is an option of threaded builds for STM (Pyboards).
